src/app/integrations/edit-page/step-configure/filter-steps/basic-filter.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component, OnInit, OnDestroy, ChangeDetectorRef } from '@angular/core';
+import { FormGroup, FormArray, FormBuilder, Validators } from '@angular/forms';
+import { ActivatedRoute, Params, Router } from '@angular/router';
+import { Subscription } from 'rxjs/Subscription';
+
+import { FlowPage } from '../../flow-page';
+import { StepStore } from '../../../../store/step/step.store';
+import { CurrentFlow, FlowEvent } from '../../current-flow.service';
+import { Action, Step } from '../../../../model';
+import { log, getCategory } from '../../../../logging';
+import { Filter } from './filter.interface';
+
+@Component({
+  selector: 'syndesis-basic-filter',
+  templateUrl: 'basic-filter.component.html',
+  styleUrls: ['./basic-filter.component.scss'],
+})
+export class BasicFilterComponent extends FlowPage implements OnInit, OnDestroy {
+  routeSubscription: Subscription;
+  position: number;
+  step: Step = <Step>{};
+  formGroup: FormGroup;
+  formConfig: any;
+  cfg: any = undefined;
+  myForm: FormGroup;
+  conditions = ['Contains', 'Does Not Contain', 'Matches Regex', 'Does Not Match Regex', 'Starts With', 'Ends With'];
+  matches = ['all', 'any'];
+
+  constructor(public currentFlow: CurrentFlow,
+    public route: ActivatedRoute,
+    public router: Router,
+    private _fb: FormBuilder,
+    public detector: ChangeDetectorRef,
+    public stepStore: StepStore) {
+    super(currentFlow, route, router, detector);
+  }
+
+  ngOnInit() {
```

Comment:
Yeah, I think maybe it'd be better to build the form from the data in the first place.  I think the format of the step is known, so I think this would be a good fit for using the ng2-dynamic-forms API which hides a lot of boilerplate (especially in the templates) that you normally have to write when using the ReactiveForms module directly.

Best fit seems to be the form array -> https://github.com/udos86/ng2-dynamic-forms#form-arrays